midend/tableHit.h

```diff
@@ -35,13 +35,26 @@ This may be needed by some back-ends which only support hit test in conditionals
 */
 class DoTableHit : public Transform, public ResolutionContext {
     TypeMap *typeMap;
+    enum op_t { None, And, Or, Xor };
+
+    const IR::Node *process(IR::AssignmentStatement *statement, op_t op);
+    const IR::Node *postorder(IR::AssignmentStatement *statement) override {
+        return process(statement, None);
+    }
+    const IR::Node *postorder(IR::OpAssignmentStatement *statement) override { return statement; }
+    const IR::Node *postorder(IR::BAndAssign *statement) override {
+        return process(statement, And);
+    }
+    const IR::Node *postorder(IR::BOrAssign *statement) override { return process(statement, Or); }
+    const IR::Node *postorder(IR::BXorAssign *statement) override {
+        return process(statement, Xor);
+    }
```

Comment:
While this works, I think it makes sense to have the `preorder` functions in the `public` part of the class.

Reply:
Well, they should never be called directly by anyone -- they should only be called from the `apply_visitor_preorder` functions in the IR classes.  So the base preorder virtual methods need to be public, but no derived visitor subclass ones do.